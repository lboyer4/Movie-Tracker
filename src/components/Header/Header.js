import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logOut } from '../../actions';

export const Header = (props) => {
	let toggleLogin;
	let logUserOut = 
		<button className='logout btn' onClick={ props.logOut }> 		
			<NavLink to='/login' className="nav"> Logout </NavLink>
		</button>
	let logUserIn = 
		<button className='login btn'>
			<NavLink to='/login' className="nav"> Login </NavLink>
		</button>
		toggleLogin = props.loggedIn.id ? logUserOut : logUserIn 
		let viewFavorites = 
		<button className='view-fave btn'>
			<NavLink to='/favorites' className="nav"> View Favorites </NavLink>
		</button>
		return (
			<header>
				<h1 className='title'> Movie Tracker </h1>
				<section className='link-wrapper'>
					{ props.loggedIn.id && viewFavorites }
					{ toggleLogin }
					<button className='signup btn'>
						<NavLink to='/signup' className="nav"> Signup </NavLink>
					</button>
				</section>
			</header>
			)
	}

export const mapStateToProps = (state) => ({
	loggedIn: state.loggedIn
});

export const mapDispatchToProps = (dispatch) => ({
	logOut: () => dispatch(logOut())
});


export default connect(mapStateToProps, mapDispatchToProps)(Header)