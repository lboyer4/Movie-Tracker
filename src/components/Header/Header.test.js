import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './Header';
import { logOut } from '../../actions';

const mockLoggedIn = {
  id: 2,
  name: "Ryan",
  password: "puppies",
  email: "puppies@gmail.com"
}

const mockLogOut= jest.fn()


describe('Header', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Header
        loggedIn={mockLoggedIn}
        logOut={mockLogOut} 
      />
    )
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapStateToProps', () => {
    it ('should return a props object with loggedIn', () => {
      const mockState = {
        loggedIn: 'loggedIn'
      }
        const expected = {
          loggedIn: 'loggedIn'
        }

        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
      });
    });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapStateToDispatch', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logOut();

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.logOut();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});