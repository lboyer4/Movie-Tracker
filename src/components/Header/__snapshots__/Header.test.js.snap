// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header>
  <h1
    className="title"
  >
     Movie Tracker 
  </h1>
  <section
    className="link-wrapper"
  >
    <button
      className="view-fave btn"
    >
      <NavLink
        className="nav"
        to="/favorites"
      >
         View Favorites 
      </NavLink>
    </button>
    <button
      className="logout btn"
      onClick={[MockFunction]}
    >
      <NavLink
        className="nav"
        to="/login"
      >
         Logout 
      </NavLink>
    </button>
    <button
      className="signup btn"
    >
      <NavLink
        className="nav"
        to="/signup"
      >
         Signup 
      </NavLink>
    </button>
  </section>
</header>
`;
